flexget/components/sites/sites/filelist_api.py

```diff
@@ -0,0 +1,180 @@
+from loguru import logger
+from flexget import plugin
+from flexget.config_schema import one_or_more
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.utils.requests import TimedLimiter, RequestException
+from flexget.utils.requests import Session as RequestSession
+
+logger = logger.bind(name='filelist')
+requests = RequestSession()
+
+
+class SearchFileList(object):
+    """
+        FileList search plugin w/ API w00t.
+    """
+
+    api_url = 'https://filelist.io/api.php'
+
+    # http error codes
+    error_codes = {
+        400: 'Invalid search/filter',
+        401: 'Username and passkey cannot be empty',
+        403: 'Invalid passkey/username',
+        429: 'Rate limit reached',
+        503: 'Service unavailable',
+    }
+
+    valid_categories = {
+        'Anime': '24',
+        'Audio': '11',
+        'Desene': '15',
+        'Diverse': '18',
+        'Docs': '16',
+        'FLAC': '5',
+        'Filme 3D': '25',
+        'Filme 4K': '6',
+        'Filme 4K Blu-Ray': '26',
+        'Filme Blu-Ray': '20',
+        'Filme DVD': '2',
+        'Filme DVD-RO': '3',
+        'Filme HD': '4',
+        'Filme HD-RO': '19',
+        'Filme SD': '1',
+        'Jocuri Console': '10',
+        'Jocuri PC': '9',
+        'Linux': '17',
+        'Mobile': '22',
+        'Programe': '8',
+        'Seriale 4K': '27',
+        'Seriale HD': '21',
+        'Seriale SD': '23',
+        'Sport': '13',
+        'TV': '14',
+        'Videoclip': '12',
+        'XXX': '7',
+    }
+
+    valid_extras = {'internal': 1, 'moderated': 1, 'freeleech': 1, 'doubleup': 1}
+    schema = {
+        'type': 'object',
+        'properties': {
+            'username': {'type': 'string'},
+            'passkey': {'type': 'string'},
+            'category': one_or_more({'type': 'string', 'enum': list(valid_categories.keys())}),
+            'internal': {'type': 'boolean', 'default': False},
+            'moderated': {'type': 'boolean', 'default': False},
+            'freeleech': {'type': 'boolean', 'default': False},
+            'doubleup': {'type': 'boolean', 'default': False},
+        },
+        'required': ['username', 'passkey'],
+        'additionalProperties': False,
+    }
+
+    def get(self, url, params):
+
+        try:
+            response = requests.get(url, params=params, raise_status=False)
+        except RequestException as e:
+            raise plugin.PluginError(f'FileList request failed badly! {e}')
+
+        if response.status_code in self.error_codes.keys():
```

Comment:
You can check `if not response.ok`. Also, there's no need to map these statuses, python has that built in:https://docs.python.org/3/library/http.html#http.HTTPStatus

Reply:
I am mapping the error codes, because those are specific to the implementation of the API. I saw that in most case the API will also return an error message, however not always so the error_codes dict holds some generic message.